core/src/main/java/org/infinispan/globalstate/LocalConfigurationManager.java

```diff
@@ -0,0 +1,51 @@
+package org.infinispan.globalstate;
+
+import java.util.EnumSet;
+
+import org.infinispan.commons.api.CacheContainerAdmin;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.manager.EmbeddedCacheManager;
+
+/**
+ * The {@link LocalConfigurationManager} is responsible for applying on each node the configuration changes initiated
+ * through the {@link org.infinispan.globalstate.GlobalConfigurationManager}.
+ *
+ * @author Tristan Tarrant
+ * @since 9.2
+ */
+
+public interface LocalConfigurationManager {
+   /**
+    * Initialization entry point for the {@link LocalConfigurationManager}
+    * @param embeddedCacheManager
+    * @param globalConfigurationManager
+    */
+   void initialize(EmbeddedCacheManager embeddedCacheManager, GlobalConfigurationManager globalConfigurationManager);
```

Comment:
I'm not sure if it is needed since you can use `@Inject` annotation to set dependencies...

Reply:
I'm not treating it as a component in the component registry, and explicitly initializing it in GlobalConfigurationManagerImpl.postStart()